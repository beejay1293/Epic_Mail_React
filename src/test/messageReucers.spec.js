import {
  UNAUTHENTICATED,
  INBOX_SUCCESS,
  SENT_SUCCESS,
  UNREAD_SUCCESS,
  DASHBOARD_STATE,
  SPECIFIC_MESSAGE_SUCCESS,
  LOADING,
  SENDING,
  MESSAGE_SUCCESS,
  MESSAGE_ERROR
} from '../constant/actionTypes';
import messageReducer, { initialState } from '../reducers/messages';


describe('test message reducers', () => {
  it('should test UNAUTHENTICATE', () => {
    expect(messageReducer(initialState, {
      type: UNAUTHENTICATED
    })).toMatchSnapshot();
  })

  it('should test INBOX_SUCCESS', () => {
    expect(messageReducer(initialState, {
      type: INBOX_SUCCESS
    })).toMatchSnapshot();
  })

  it('should test SENT_SUCCESS', () => {
    expect(messageReducer(initialState, {
      type: SENT_SUCCESS
    })).toMatchSnapshot();
  })

  it('should test UNREAD_SUCCESS', () => {
    expect(messageReducer(initialState, {
      type: UNREAD_SUCCESS
    })).toMatchSnapshot();
  })

  it('should test DASHBOARD STATE', () => {
    expect(messageReducer(initialState, {
      type: DASHBOARD_STATE
    })).toMatchSnapshot();
  })
  it('should test LOADING', () => {
    expect(messageReducer(initialState, {
      type: LOADING
    })).toMatchSnapshot();
  })
  it('should test SENDING', () => {
    expect(messageReducer(initialState, {
      type: SENDING
    })).toMatchSnapshot();
  })
  it('should test MESSAGE_SUCCESS', () => {
    expect(messageReducer(initialState, {
      type: MESSAGE_SUCCESS
    })).toMatchSnapshot();
  })
})