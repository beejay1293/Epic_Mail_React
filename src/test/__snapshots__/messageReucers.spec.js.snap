// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test message reducers should test DASHBOARD STATE 1`] = `
Object {
  "body": undefined,
  "dashboardstate": undefined,
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test INBOX_SUCCESS 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": undefined,
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test LOADING 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": true,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test MESSAGE_SUCCESS 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "message sent successfully",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test SENDING 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": true,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test SENT_SUCCESS 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": undefined,
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test UNAUTHENTICATE 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": Array [],
}
`;

exports[`test message reducers should test UNREAD_SUCCESS 1`] = `
Object {
  "body": "viewMessages",
  "dashboardstate": "Inbox",
  "errors": Object {},
  "inbox": Array [],
  "isAuthenticated": false,
  "isLoading": false,
  "messageError": "",
  "messageSuccess": "",
  "receiver": Object {},
  "sender": Object {},
  "sending": false,
  "sent": Array [],
  "specificMessage": Object {},
  "unread": undefined,
}
`;
